### PR TITLE
created book service and repository

### PR DESCRIPTION
**Pull request by KhrystynaDidykh**

### CODE REVIEW

File: src/main/java/org/example/bookapp/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookapp/model/Book.java]: image
Put the validation annotations

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1 +1,11 @@
...
+spring.datasource.url=jdbc:mysql://localhost:3306/cinemaNew?serverTimeZone=UTC
+spring.datasource.username=root
+spring.datasource.password=202020xxx
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
+
+
+
```
Comment (MAMentorFX) [inline src/main/resources/application.properties]:
```suggestion
```

File: src/main/java/org/example/bookapp/BookAppApplication.java [hunk 3]
```diff
@@ -1,13 +1,39 @@
...
 
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
+            @Override
+            public void run(String... args) throws Exception {
+                Book book = new Book();
+                book.setTitle("Ракета на чотирьох лапах ");
+                book.setAuthor("Джеремі Стронґ");
+                book.setDescription("книга про пригоди");
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookapp/BookAppApplication.java]: Please don't use any Cyrillic

File: src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.example.bookapp.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
```
Comment (MAMentorFX) [inline src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java]:
```suggestion
```

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

File: src/checkstyle.xml [hunk 5]
```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0"?>
```
Comment (Elena-Bruyako) [inline src/checkstyle.xml]: This file shouldn't be in src folder

File: src/main/java/org/example/bookapp/model/Book.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookapp/model/Book.java]: Don't use `@Data` for entities 
Add `@Table `

File: src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java]: Don't use `@Autowired` if you have only one constructor
Add `@RequiredArgsConstructor`

File: src/main/java/org/example/bookapp/service/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import org.example.bookapp.model.Book;
+import org.example.bookapp.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookapp/service/BookServiceImpl.java]: The same

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/org/example/bookapp/model/Book.java [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+        this.description = description;
+    }
+
+    public String getCoverImage() {
+        return coverImage;
+    }
+
+    public void setCoverImage(String coverImage) {
+        this.coverImage = coverImage;
+    }
```
Comment (koval-kn) [inline src/main/java/org/example/bookapp/model/Book.java]: using `@Getter / @Setter` is allowed with `@Entity`, so better replace this boilerplate code

File: src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (koval-kn) [inline src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java]: just use `@RequiredArgsConstructor`
```suggestion
```

File: src/main/java/org/example/bookapp/service/BookServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+import org.example.bookapp.repository.BookRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (koval-kn) [inline src/main/java/org/example/bookapp/service/BookServiceImpl.java]: just use `@RequiredArgsConstructor`
```suggestion
```

File: src/main/resources/application.properties [hunk 12]
```diff
@@ -1 +1,8 @@
 spring.application.name=bookApp
+spring.datasource.url=jdbc:mysql://localhost:3306/cinemaNew?serverTimeZone=UTC
```
Comment (koval-kn) [inline src/main/resources/application.properties]: for the future - SQL dont cares about registry case, so while naming, better to prefer snake_case over camelCase
```suggestion
spring.datasource.url=jdbc:mysql://localhost:3306/cinema_new?serverTimeZone=UTC
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/bookapp/model/Book.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    private String isbn;
+    @Getter
+    @Setter
+    @Column(nullable = false)
+    private BigDecimal price;
+    @Getter
+    @Setter
+    private String description;
+    @Getter
+    @Setter
```
Comment (koval-kn) [inline src/main/java/org/example/bookapp/model/Book.java]:
```suggestion
@Getter
@Setter
@Entity
@Table(name = "books")
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String title;
    @Column(nullable = false)
    private String author;
    @Column(unique = true, nullable = false)
    private String isbn;
    @Column(nullable = false)
    private BigDecimal price;
    private String description;
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 12 above)
Comment (Elena-Bruyako) [reply src/main/resources/application.properties]: Not fixed

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by MAMentorFX — APPROVED** [review]

**Comment by KhrystynaDidykh** [comment]
When I add @RequiredArgsConstructor or @Getter @Setter my program doesn't work, how can fix it?

**Comment by Elena-Bruyako** [comment]
> When I add @requiredargsconstructor or @getter @Setter my program doesn't work, how can fix it?

Provide the stack trace to clarify what to you mean "program doesn't work". Let's discuss it in chat on the platform

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
See the comment

**Pull request merged**